### PR TITLE
Add ebpf proxy mode support for Cilium CNI

### PR DESCRIPTION
**Pull request by rastislavs**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Add ebpf proxy mode support for Cilium CNI
```

### CODE REVIEW

File: addons/kube-proxy/serviceaccount.yaml [hunk 1]
```diff
@@ -12,8 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+{{ if ne .Cluster.Network.ProxyMode "ebpf" }}
```
Comment (wozniakjan) [inline addons/kube-proxy/serviceaccount.yaml:15]: I just hope someone from the kube-proxy team doesn't implement ebpf backend for it too.. :)

**Review by wozniakjan — APPROVED** [review]
/lgtm

File: addons/kube-proxy/serviceaccount.yaml (hunk 1 above)
Comment (rastislavs) [reply addons/kube-proxy/serviceaccount.yaml:15]: then we change it to the `cni` proxy mode :)

Comment (rastislavs) [reply addons/kube-proxy/serviceaccount.yaml:15]: .. or add CNI type into the condition

**Pull request merged**